storage/mysql/tree_storage.go

```diff
@@ -19,19 +19,24 @@ import (
 	"context"
 	"database/sql"
 	"encoding/base64"
+	"flag"
 	"fmt"
 	"runtime/debug"
 	"strings"
 	"sync"
+	"time"
 
 	"github.com/golang/glog"
 	"github.com/golang/protobuf/proto"
 	"github.com/google/trillian"
 	"github.com/google/trillian/storage"
 	"github.com/google/trillian/storage/cache"
 	"github.com/google/trillian/storage/storagepb"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
+var maxConns = flag.Int("max_db_conns", 1000, "Maximum connections to the database")
```

Comment:
We'd ideally prefer to have flag definitions only in `main` packages, not libraries, and inject their values as parameters instead. 

In this case, I don't think there's a way to percolate the flag all the way up to `main`, but I think it could live in `server/mysql_storage_provider.go` (in an analogous manner to the flags in `server/etcd_quota_provider.go`) ?